pycbc/events/significance.py

```diff
@@ -0,0 +1,253 @@
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+
+"""
+This module contains functions to calculate the significance
+through different estimation methods of the background, and functions that
+read in the associated options to do so.
+"""
+import logging
+import copy
+import numpy as np
+from pycbc.events import coinc
+from pycbc.events import trigger_fits as trstats
+
+
+def n_louder(bstat, fstat, dec, **kwargs):  # pylint:disable=unused-argument
+    """
+    Count the number of louder background events, taking decimation
+    into account
+    """
+    return coinc.calculate_n_louder(bstat, fstat, dec)
+
+
+def trig_fit(back_stat, fore_stat, dec_facs, fit_func=None,
+             fit_thresh=None):
+    """
+    Use a fit to events in back_stat in order to estimate the
+    distribution for use in recovering the ifars. Below the
+    fit threshold, use the n_louder method for these triggers
+    """
+    # Manually set some defaults:
+    fit_func = fit_func if fit_func else 'exponential'
+    fit_thresh = fit_thresh if fit_thresh else 0
+
+    # Calculate the fitting factor of the ranking statistic distribution
+    alpha, _ = trstats.fit_above_thresh(fit_func, back_stat,
+                                        thresh=fit_thresh,
+                                        weights=dec_facs)
+
+    # Count background events above threshold as the cum_fit is
+    # normalised to 1
+    bg_above = back_stat > fit_thresh
+    bg_above_thresh = np.sum(dec_facs[bg_above])
+    fg_above = fore_stat > fit_thresh
+
+    # These will be overwritten, but just to silence a warning
+    # in the case where trstats.cum_fit returns zero
+    back_cnum = np.zeros_like(back_stat)
+    fnlouder = np.zeros_like(fore_stat)
+
+    # Ue the fit above the threshold
+    back_cnum[bg_above] = trstats.cum_fit(fit_func, back_stat[bg_above],
+                                          alpha, fit_thresh) * bg_above_thresh
+    fnlouder[fg_above] = trstats.cum_fit(fit_func, fore_stat[fg_above],
+                                         alpha, fit_thresh) * bg_above_thresh
+
+    # below the fit threshold, we count the number of louder events,
+    # as things get complicated by clustering below this point
+    fg_below = np.logical_not(fg_above)
+    bg_below = np.logical_not(bg_above)
+
+    back_cnum[bg_below], fnlouder[fg_below] = \
+        n_louder(back_stat, fore_stat, dec_facs)
+
+    return back_cnum, fnlouder
+
+
+_significance_meth_dict = {
+    'trigger_fit': trig_fit,
+    'n_louder': n_louder
+}
+
+
+def calculate_n_louder(method_dict, back_stat, fore_stat, dec_facs):
```

Comment:
This function shadows one in coinc.py in naming. Can you use a different name here that reflects how this is different? Also, why is a method_dist passed instead of kwargs?

Reply:
Maybe 'estimate'?

Reply:
@ahnitz I have just started reviewing this PR myself, it will be confusing if we're both doing it at once. How do you want to do this?

Reply:
@tdent I give up the token. Go ahead. I'll say no more.

Reply:
OK, I will try and remember to give it back ..

Reply:
in any case, I agree this function should have a different name .. as it is just a wrapper I would suggest something like `get_n_louder`